tests/unit/test_semantic_layer_nodes_satisfy_protocols.py

```diff
@@ -0,0 +1,123 @@
+from dbt.contracts.graph.nodes import (
+    Metric,
+    MetricInputMeasure,
+    MetricTypeParams,
+    NodeRelation,
+    SemanticModel,
+    WhereFilter,
+)
+from dbt.contracts.graph.semantic_models import Dimension, DimensionTypeParams, Entity, Measure
+from dbt.node_types import NodeType
+from dbt_semantic_interfaces.protocols.dimension import Dimension as DSIDimension
+from dbt_semantic_interfaces.protocols.entity import Entity as DSIEntitiy
+from dbt_semantic_interfaces.protocols.measure import Measure as DSIMeasure
+from dbt_semantic_interfaces.protocols.metric import Metric as DSIMetric
+from dbt_semantic_interfaces.protocols.semantic_model import SemanticModel as DSISemanticModel
+from dbt_semantic_interfaces.type_enums.dimension_type import DimensionType
+from dbt_semantic_interfaces.type_enums.entity_type import EntityType
+from dbt_semantic_interfaces.type_enums.metric_type import MetricType
+from dbt_semantic_interfaces.type_enums.time_granularity import TimeGranularity
+from typing import Protocol, runtime_checkable
+
+
+@runtime_checkable
+class RuntimeCheckableSemanticModel(DSISemanticModel, Protocol):
+    pass
+
+
+@runtime_checkable
+class RuntimeCheckableDimension(DSIDimension, Protocol):
+    pass
+
+
+@runtime_checkable
+class RuntimeCheckableEntity(DSIEntitiy, Protocol):
+    pass
+
+
+@runtime_checkable
+class RuntimeCheckableMeasure(DSIMeasure, Protocol):
+    pass
+
+
+@runtime_checkable
+class RuntimeCheckableMetric(DSIMetric, Protocol):
+    pass
+
+
+def test_semantic_model_node_satisfies_protocol():
+    test_semantic_model = SemanticModel(
```

Comment:
It's awkward that we have to instantiate dummy classes to do this check. It makes the tests verbose and could be a minor maintenance issue. I think we could probably come up with a more compact and flexible way of doing these checks.

Reply:
How would it be a maintenance issue? The main thing I can think of is if the DSI version is incompatible, then the tests would start failing. However, that's kind of what we want, right?